streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -201,13 +210,24 @@ public MockProcessorContext(final Properties config) {
      */
     @SuppressWarnings({"WeakerAccess", "unused"})
     public MockProcessorContext(final Properties config, final TaskId taskId, final File stateDir) {
-        final StreamsConfig streamsConfig = new StreamsConfig(config);
+        final StreamsConfig streamsConfig = new InternalStreamsConfig(config, false);
         this.taskId = taskId;
         this.config = streamsConfig;
         this.stateDir = stateDir;
         this.metrics = new StreamsMetricsImpl(new Metrics(), "mock-processor-context-virtual-thread");
     }
 
+    private static class InternalStreamsConfig extends StreamsConfig {
+
+        public InternalStreamsConfig(final Map<?, ?> props) {
+            super(props);
+        }
+
+        protected InternalStreamsConfig(final Map<?, ?> props, final boolean doLog) {
+            super(props, doLog);
+        }
+    }
+
```

Comment:
just to quick spamming the logs.

Reply:
which I think is reasonable, since this context would only be used for unit tests.

Reply:
There is already an `InternalStreamsConfig` in `StreamsPartitionsAssignor` -- if we want to use it multiple time, should we make it a public internal call that we can reuse here instead of duplicating code? Maybe in a new package `...streams.internals` ?

Reply:
I considered that, but it seemed better to keep the MockProcessorContext in test-utils as decoupled as possible from the internals of Streams.

In general, when we're modifying Streams code, we exercise a lot of freedom in modifying internals, and I don't want to risk accidentally changing the behavior of the mock if we decide to add some more stuff to the one used by StreamsPartitionAssigner.

If it helps, I can give this one a different name...